pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -981,59 +994,88 @@ func (a *Operator) updateNamespaceList(op *operatorsv1.OperatorGroup) ([]string,
 	return namespaceList, nil
 }
 
-func (a *Operator) ensureOpGroupClusterRole(op *operatorsv1.OperatorGroup, suffix string, apis cache.APISet) error {
-	clusterRole := &rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   strings.Join([]string{op.GetName(), suffix}, "-"),
-			Labels: map[string]string{install.OLMManagedLabelKey: install.OLMManagedLabelValue},
-		},
+func (a *Operator) getClusterRoleName(op *operatorsv1.OperatorGroup, roleType string) (string, error) {
+	roleSuffix := hash(fmt.Sprintf("%s/%s/%s", op.GetNamespace(), op.GetName(), roleType))
+	// calculate how many characters are left for the operator group name
+	nameLimit := kubeResourceNameLimit - len(strings.Replace(operatorGroupClusterRoleNameFmt, "%s", "", -1)) - len(roleType) - len(roleSuffix)
+	if len(op.GetName()) < nameLimit {
+		return fmt.Sprintf(operatorGroupClusterRoleNameFmt, op.GetName(), roleType, roleSuffix), nil
 	}
-	var selectors []metav1.LabelSelector
-	for api := range apis {
-		aggregationLabel, err := aggregationLabelFromAPIKey(api, suffix)
-		if err != nil {
-			return err
-		}
-		selectors = append(selectors, metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				aggregationLabel: "true",
-			},
-		})
-	}
-	if len(selectors) > 0 {
-		clusterRole.AggregationRule = &rbacv1.AggregationRule{
-			ClusterRoleSelectors: selectors,
-		}
+	return fmt.Sprintf(operatorGroupClusterRoleNameFmt, op.GetName()[:nameLimit], roleType, roleSuffix), nil
+}
+
+func (a *Operator) ensureOpGroupClusterRole(op *operatorsv1.OperatorGroup, suffix string, apis cache.APISet) error {
+	// create target cluster role spec
+	var clusterRole *rbacv1.ClusterRole
+	clusterRoleName, err := a.getClusterRoleName(op, suffix)
+	if err != nil {
+		return err
 	}
-	err := ownerutil.AddOwnerLabels(clusterRole, op)
+	aggregationRule, err := a.getClusterRoleAggregationRule(apis, suffix)
 	if err != nil {
 		return err
 	}
 
-	existingRole, err := a.lister.RbacV1().ClusterRoleLister().Get(clusterRole.Name)
+	// get existing cluster role for this level (suffix: admin, edit, view))
+	existingRole, err := a.lister.RbacV1().ClusterRoleLister().Get(clusterRoleName)
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	if apierrors.IsNotFound(err) {
-		existingRole, err = a.opClient.KubernetesInterface().RbacV1().ClusterRoles().Create(context.TODO(), clusterRole, metav1.CreateOptions{})
-		if err == nil {
+
+	if existingRole != nil {
+		// if the existing role conforms to the naming convention, check for skew
```

Comment:
Prefer `semantic.Equality` for checks on k8s objects. Sometimes thing like order in arrays matters.

Reply:
Or, alternatively, simply use server-side apply to assert that the state you want for the object is right. k8s server will do any diffs you need.

Reply:
Does this block this PR?

Reply:
Yes

Reply:
updating using `equality.Semantic` - I haven't used SSA. I'd need a bit more time. If you want I can create an issue to move to SSA.

Reply:
That's fine for now. SSA issue for the future would be good. More important that we do it in every case for v1 than we go back to improve every possible case in v0.